### ISSUE TITLE
KeyError: 'proxy-authenticate' - response2 gives real content instead of challenge

### ISSUE THREAD

**Issue by orbidlo** [requests/requests-ntlm#49]
TLDR: Response2 gives real content instead of challenge. Not sure if broken proxy or false assumption.

Possibly related with https://github.com/requests/requests-ntlm/issues/36

I'm behind evil corporate NTLM proxy. While within session I would request a page (successfully), grab it's content to get form postdata and finally send new request which would end up with this error:

```
res = s.post(url, data=formdata, stream=True)
File "C:\Python2\lib\site-packages\requests\sessions.py", line 498, in post
return self.request('POST', url, data=data, **kwargs)
File "C:\Python2\lib\site-packages\requests\sessions.py", line 456, in request
resp = self.send(prep, **send_kwargs)
File "C:\Python2\lib\site-packages\requests\sessions.py", line 565, in send
r = dispatch_hook('response', hooks, r, **kwargs)
File "C:\Python2\lib\site-packages\requests\hooks.py", line 41, in dispatch_hook
_hook_data = hook(hook_data, **kwargs)
File "C:\Python2\lib\site-packages\requests_ntlm\requests_ntlm.py", line 126, in response_hook
kwargs)
File "C:\Python2\lib\site-packages\requests_ntlm\requests_ntlm.py", line 91, in retry_using_http_NTLM_auth
auth_header_value = response2.headers[auth_header_field]
File "C:\Python2\lib\site-packages\requests\structures.py", line 77, in __getitem__
return self._store[key.lower()][1]
KeyError: 'proxy-authenticate'
```

With the help of a friend I digged into it and found out, that request gets 407 response, hook is dispatched and caught by response_hook to call retry_using_http_NTLM_auth(). That function makes assumption that it needs to first send NTLM negotiate message, receives challenge in response2, builds new request with NTLM authenticate and finally gets real content in response3.

Instead of challenge, what I really got in response2 was real content. Response2.header thus did not contain any key 'proxy-authenticate' (auth_header_value) hence the error. I made a dirty but working workaround and changed + added code from line 91:

```
auth_header_value = response2.headers.get(auth_header_field) 
if not auth_header_value:
    return response2
```

I'm really unsure why our proxy behaves like this - could be just bad implementation that breaks some NTLM protocol rule (please let me know, if it does, so I can yell at our IT) or just wrong assumption from the author/s. I'm not very good programmer hence why I created issue and not pull request. Hopefully you can dig into this and create better patch (or tell me to yell at our IT instead).
